### PR TITLE
Introduce batches in BatchInserterImpl, prevent empty property records emergence on node properties update 

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Prevent emergence of empty property records during node properties update.
Update BatchInserterImpl to perform commits in batches. Remove pool from DirectRecordAccess, switch batch backing map to map with constant-time performance for get and put operations.

### CODE REVIEW

**Comment by tinwelint** [comment]
You should squasch those two commits into one, the second doesn't make sense on its own

**Comment by tinwelint** [comment]
I think this should be fixed for 2.2 or even earlier, but perhaps 2.2 is fine.

File: community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java [hunk 1]
```diff
@@ -342,17 +347,27 @@ public boolean relationshipHasProperty( long relationship, String propertyName )
...
     public void setRelationshipProperty( long relationship, String propertyName, Object propertyValue )
     {
-        propertyCreator.setPrimitiveProperty( getRelationshipRecord( relationship ),
-                getOrCreatePropertyKeyId( propertyName ), propertyValue, recordAccess.getPropertyRecords() );
-        recordAccess.commit();
+        RecordProxy<Long,RelationshipRecord,Void> relationshipRecord = getRelationshipRecord( relationship );
+        int propertyKey = getOrCreatePropertyKeyId( propertyName );
+        RecordAccess<Long,PropertyRecord,PrimitiveRecord> propertyRecords = recordAccess.getPropertyRecords();
+
+        removePropertyIfExist( relationshipRecord, propertyKey, propertyRecords );
```
Comment (tinwelint) [inline community/kernel/src/main/java/org/neo4j/unsafe/batchinsert/BatchInserterImpl.java]: Setting the property this way and given the changes in `PropertyCreator` makes me think that the `PropertyCreator#setPrimitiveProperty` is completely unnecessary. So instead just use `primitiveAddProperty` and remove `setPrimitiveProperty`.

The drawback of your changes is that the property chain needs to be traversed potentially two times, first to find-and-delete and then to find space to allocate the new one. The purpose of `PropertyCreator#setPrimitiveProperty` was to do that in a single method, with only one chain traversal.

**Comment by MishaDemianenko** [comment]
Was expecting some comments to come :), squashed, update code to use  `primitiveAddProperty` and clean `PropertyCreator`

**Pull request closed without merging**